test/functional/feature_config_args.py

```diff
@@ -30,8 +30,8 @@ def test_config_file_parser(self):
         self.nodes[0].assert_start_raises_init_error(expected_msg='Error reading configuration file: parse error on line 1: nono, if you intended to specify a negated option, use nono=1 instead')
 
         with open(inc_conf_file_path, 'w', encoding='utf-8') as conf:
-            conf.write('server=1\nrpcuser=someuser\nrpcpassword=some#pass')
-        self.nodes[0].assert_start_raises_init_error(expected_msg='Error reading configuration file: parse error on line 3, using # in rpcpassword can be ambiguous and should be avoided')
+            conf.write('server=1\nrpcuser=someuser\n[main]\nrpcpassword=some#pass')
```

Comment:
maybe keep one outside the section too

Reply:
okay, it's clear we need to test all three, after the recent discussion it is very clear it's easy to miss one:
```
rpcpassword=   # bare
regtest.rpcpassword=   # explicit section

[regtest]
rpcpassword=   # implicit section
```